api_v3/services/BaseEntryService.php

```diff
@@ -26,6 +31,47 @@ public function initService($serviceId, $serviceName, $actionName)
             throw new KalturaAPIException(KalturaErrors::ACTION_FORBIDDEN, "anonymousRank");
         }
     }
+	
+	public function adjustArguments(&$arguments = null, $actionParams = null)
+	{
+		$params = requestUtils::getRequestParams();
+		
+		$skipDeserializer = $this->adjustParameters($params);
+		
+		if (!$skipDeserializer)
+		{
+			$deserializer = new KalturaRequestDeserializer($params);
+			$arguments = $deserializer->buildActionArguments($actionParams);
+			
+			KalturaLog::debug("Dispatching service [" . $this->serviceName . "], action [" . $this->actionName . "], reqIndex [" .
+			                  kCurrentContext::$multiRequest_index . "] with ADJUSTED params " . print_r($arguments, true));
+			
+			$responseProfile = $deserializer->getResponseProfile();
+			if ($responseProfile)
+			{
+				KalturaLog::debug("Response profile: " . print_r($responseProfile, true));
+				$this->setResponseProfile($responseProfile);
+			}
+		}
+	}
+	
+	protected function adjustParameters(&$params)
+	{
+		$skipDeserializer = true;
+		$supportedFields = entry::getMultiLingualSupportedFields();
```

Comment:
this can be in some utility and since we implemented getMultiLingualSupportedFields in the interface you can call it from the general object and not specifically on entry